hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java

```diff
@@ -298,6 +298,8 @@ public void markContainerForClose() throws StorageContainerException {
   public void markContainerUnhealthy() throws StorageContainerException {
     writeLock();
     try {
+      // TODO: container should be marked UNHEALTHY in memory even if container
```

Comment:
Good point. But, I think this is how it is already implemented. updateContainerData() on encountering an exception, does not reset the state if the state is UNHEALTHY. Or did I get that wrong?

Reply:
You are right. I missed the state check.